src/core/server/elasticsearch/cluster_client.ts

```diff
@@ -106,14 +88,27 @@ export interface FakeRequest {
   headers: Headers;
 }
 
+/**
+ * {@inheritdoc ClusterClient}
```

Comment:
There's no good options, but it feels like we should try to standardise on our approach here.

`@inheritdoc` comments aren't supported by VSCode, so hovering over `IClusterClient` or any methods on this type isn't very useful. One option is to move all the doc comments to the interface and inherit them from the class. I don't like that so much because then the documentation comments are separated from their implementation.

<img width="687" alt="Screenshot 2019-10-02 at 14 43 13" src="https://user-images.githubusercontent.com/1281646/66045068-4114a980-e523-11e9-8080-5dbb81a56149.png">

I've been using the `PublicMethodsOf<InstanceClass>` type and then adding an `{@link InstanceClass}` doc comment. It requires an extra click when going through documentation, but then we get generated docs and good tsdocs when hovering over types without sacrificing having doc comments be defined right next to the implementation code.

Reply:
One problem with both of these approaches is that we have to export the concrete class from the entire package in order to get docs generated. I don't particularly like exporting concrete classes that are unnecessary.

But I agree that your approach is probably best in order to have the docs next to the implementation. Interestingly, VSCode does support inheriting docs from parent classes and interfaces automatically. Wish it would work with explicit `{@inheritDoc Class.prop}` directives.

I'm going to prefer `Pick` over `PublicMethodsOf` to make sure that new functions added later don't get leaked in the interface if we only intend to use them inside core.